racker/salus-telemetry-resource-management
# Swagger documentation

# Resolves

    SALUS-318

# What

Add documentation for our API endpoints

# How

We need to generate the documentation on the internal API's (resource management, monitor management, etc...) but they have slightly different paths compared to the public API.

## How to test

pull this branch and run the Compile Maven command. (you need to compile so that the annotations get compiled into the source for picked by the maven swagger plugin). You should see output in {basedir}/target/generated/swagger/swagger.json

There is a new maven endpoint under plugins->exec->java
This new target will run the converter on the swagger json and output to {basedir}/target/generated/swagger/convertedOutput.json
This is how to generate the swagger for the public api

# Why

We can't really run the documentation processes on the public api because it just proxies all of the data through. So we basically need to create the documentation on the internal services and alter the path to reflect the public api path.

# TODO

Add a function to the mustache html template that will take the paths and create the correct path
